src/Avalonia.Controls/Border.cs

```diff
@@ -92,12 +95,63 @@ public override void Render(DrawingContext context)
 
             if (background != null)
             {
-                context.FillRectangle(background, rect, cornerRadius);
+                context.FillRectangle(Brushes.Transparent, rect);
             }
 
-            if (borderBrush != null && borderThickness > 0)
+            if (borderBrush != null)
             {
-                context.DrawRectangle(new Pen(borderBrush, borderThickness), rect, cornerRadius);
```

Comment:
Check for thickness being uniform (e.g. `(1,1,1,1)` and call `DrawRectangle` in that case), geometry drawing is slow.